### PR TITLE
SDK-2437: adding dynamic lang support and baseUrl configuration, ensuring consistant variable names

### PR DESCRIPTION
**Pull request by myrr-BeamImpact**
### All pull requests:
* [ ] Have you considered security (e.g., XSS)?
* [ ] Have you considered desktop, mobile, and tablet form-factors?
* [ ] Have you considered [accessibility best practices](https://www.w3.org/WAI/standards-guidelines/wcag/)?

### Code
* [ ] Are the commits squashed into one commit?
* [ ] Is the [branch name and commit message](CONTRIBUTING.md#conventions-for-branch-names-and-commit-messages) following team convention?
* [ ] Is the code linted?
* [ ] Are all open issues, questions, and concerns by reviewers answered, resolved, and reviewed?

### Quality assurance
* [ ] If applicable, are unit tests written and are they passing?
* [ ] If applicable, are integration tests written and are they passing?
* [ ] Have checks linked in your pull request passed?

### Documentation
* [ ] Are there meaningful code comments included?

## Applicable to community contributors:
* [ ] Have you granted SFRA team access to your fork? [Grant access](CONTRIBUTING.md#community-contributors) 

### Documentation
* [ ] If applicable, is the change log updated?
* [ ] If applicable, have any UI text changes been reviewed by Documentation?
* [ ] If applicable, have any UI implementations been reviewed by the UX team?

### Security
* [ ] If applicable, has Security reviewed this code?
* [ ] If applicable, is a 3PP request submitted?

### CODE REVIEW

File: cartridges/plugin_beam/cartridge/templates/default/beam/beam_subscription_management.isml [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
-<script type="module" src="https://sdk.beamimpact.com/web-sdk/${sitePref.version}/dist/components/subscription-management.js"></script>
+<script type="module" src="https://sdk.beamimpact.com/web-sdk/${sitePref.beamVersion}/dist/components/subscription-management.js"></script>
 
 <beam-subscription-management
     apikey="${sitePref.beamApiKey}"
     chainid="${sitePref.beamChainId}"
     storeid="${sitePref.beamStoreId}"
     email="${pdict.order.orderEmail}"
+    baseurl="${sitePref.beamBaseUrl}"
     subscriptionid="${pdict.order.orderNumber}"
```
Comment (jmastic-beam) [inline cartridges/plugin_beam/cartridge/templates/default/beam/beam_subscription_management.isml:11]: `question` Does this component not accept `lang` as a parameter?

File: metadata/beam_configuration.xml [hunk 2]
```diff
@@ -155,7 +155,7 @@
             <attribute-group group-id="Beam Credentials">
                 <display-name xml:lang="x-default">Beam Impact Widget Credentials</display-name>
                 <attribute attribute-id="beamStoreId"/>
-                <attribute attribute-id="beamLang"/>
```
Comment (jmastic-beam) [inline metadata/beam_configuration.xml:158]: `question` Was the intention to make the language ALWAYS inherit from the page, and fall back to English if it's not supported? I think keeping `beamLang` could allow a partner to specify a fallback other than English, but if that's not wanted, this seems fine.

**Review by jmastic-beam — APPROVED** [review]
Left a couple questions - the code changes seem fine as long as the answers to the questions don't require additional changes.

**Review by DerekBeam — APPROVED** [review]
Approved barring Justin's questions being answered. I also had concerns about not being able to set beam lang as a config option

**Review by DerekBeam — APPROVED** [review]

**Pull request merged**